extensions/interactions/NumberWithUnits/directives/number-with-units-rules.service.ts

```diff
@@ -47,14 +47,13 @@ export class NumberWithUnitsRulesService {
     var answerObject = this.unitsObjectFactory.fromDict(answer);
     var inputsObject = this.unitsObjectFactory.fromDict(inputs.f);
 
-    var answerString = answerObject.toMathjsCompatibleString();
-    var inputsString = inputsObject.toMathjsCompatibleString();
+    var answerNum = answerObject.toMathjsCompatibleString().split(' ')[0];
```

Comment:
It would be better if you could get the numbers programmatically rather than through indirect string manipulation. Couldn't you just take the "numerical part" of the `answer` variable?

Reply:
Done. now we are first checking whether the input and answer are of same type ie real or fraction is not which means the answer may be correct but its not of same type (2 != 4/2) and then we extract the values and compare.